azimuth_capi/operator.py

```diff
@@ -574,6 +576,20 @@ async def ensure_platform(instance: api.Cluster, realm):
     kopf.adopt(platform, instance.model_dump())
     return await ekclient.apply_object(platform, force = True)
 
+# Separate on create handler to ensure etcd key is only generated once on first install
+@model_handler(api.Cluster, kopf.on.create)
+async def generate_etcd_key(logger, instance, name, namespace, patch, **kwargs):
+    await ekclient.create_object({
```

Comment:
I think I would rather we did something more idempotent within the create and update loop, to save having to wait on each other.

Basically, just check if its there (i.e. your fetch call below) and if its missing create it.